cpp/util/ManagedArray.h

```diff
@@ -235,11 +261,7 @@ template<typename T> class ManagedArray
      */
     static inline std::vector<size_t> getMultiIndex(std::vector<size_t> shape, size_t index)
     {
-        size_t cur_prod = 1;
-        for (auto it = shape.begin(); it != shape.end(); ++it)
-        {
-            cur_prod *= *it;
-        }
+        size_t cur_prod = std::accumulate(shape.begin(), shape.end(), 1, std::multiplies<>());
```

Comment:
Use const iterators. Also does it make sense to name it "_current_" product now that it's computed in one line?

```suggestion
        size_t cur_prod = std::accumulate(shape.cbegin(), shape.cend(), 1, std::multiplies<>());
```